upload_vehicle_data.py

```diff
@@ -0,0 +1,830 @@
+        logger.error(f'Failed to run SSH command: {command}: {e}')
+        raise
+
+
+def create_remote_temp_directory(
+    logger, remote_user, remote_ip, remote_directory
+):
+    """Create a directory on the remote machine."""
+    command = f'mkdir -p {remote_directory}/temp'
+    run_ssh_command(logger, remote_user, remote_ip, command)
+    logger.debug(
+        f'Created remote temporary directory: {remote_directory}/temp'
+    )
+
+
+def delete_remote_temp_directory(
+    logger, remote_user, remote_ip, remote_directory
+):
+    """Delete a directory on the remote machine."""
+    command = f'rm -rf {remote_directory}/temp'
+    run_ssh_command(logger, remote_user, remote_ip, command)
+    logger.debug(
+        f'Deleted remote temporary directory: {remote_directory}/temp'
+    )
+
+
+def delete_remote_directory_contents(
+    logger, remote_user, remote_ip, remote_directory
+):
+    """Delete the contents of a directory on the remote machine."""
+    command = f'rm -rf {remote_directory}/*'
+    run_ssh_command(logger, remote_user, remote_ip, command)
+    logger.debug(f'Deleted contents of remote directory: {remote_directory}')
+
+
+def get_remote_home_directory(logger, remote_user, remote_ip):
+    """Get the home directory of the remote user."""
+    home_dir_cmd = f'ssh {remote_user}@{remote_ip} "eval echo ~$USER"'
+    try:
+        result = subprocess.run(
+            home_dir_cmd,
+            shell=True,
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        remote_home_directory = result.stdout.strip()
+        logger.info(
+            f'Remote home directory for {remote_user} '
+            f'is {remote_home_directory}'
+        )
+        return remote_home_directory
+    except subprocess.CalledProcessError as e:
+        logger.error(f'Failed to get remote home directory: {e}')
+        raise
+
+
+def compress_and_transfer_rosbag(
+    logger,
+    remote_user,
+    remote_ip,
+    rosbag_path,
+    remote_directory,
+    cloud_upload_directory,
+    mcap_path,
+    max_upload_attempts,
+    base_remote_directory,
+    current_rosbag_number,
+    total_rosbags,
+    global_rosbag_counter,
+):
+    """Compress rosbag on remote machine and transfer it to cloud host."""
+    remote_temp_directory = f'{remote_directory}/temp'
+    relative_bag_path = os.path.relpath(
+        rosbag_path, start=base_remote_directory
+    )
+    # Check available disk space before compression
+
+    if not check_disk_space(
+        logger, remote_user, remote_ip, remote_temp_directory, rosbag_path
+    ):
+        logger.error(f'Insufficient disk space for compressing {rosbag_path}')
+        return False
+
+    logger.info(
+        f'Enough space found on the remote machine. Start '
+        f'compressing: \n{rosbag_path}'
+    )
+
+    # Compress the rosbag on the remote machine
+    logger.info(f'Starting compression of {rosbag_path} on remote machine...')

```

Comment:
These logs are redundant, we should merge them or stick with one. I suggest:

```suggestion
    # Compress the rosbag on the remote machine
    logger.info(
        f'Enough space found on the remote machine. Start '
        f'compressing:\n{rosbag_path}'
    )
```

Reply:
Changed.